packages/kbn-esql-ast/src/types.ts

```diff
@@ -78,6 +78,17 @@ export interface ESQLLiteral extends ESQLAstBaseItem {
   value: string | number;
 }
 
+export interface ESQLStringLiteral extends ESQLLiteral {
```

Comment:
I would prefer to keep types as minimal as possible here as they are publicly exposed from the package, and adding all sort of custom type will just confuse the final user of the package.

I can see few alternatives here:
* define `ESQLLiteral` as union of types (based on values I'd say 2 subTypes should be enough)
* cast `value` as `string` and push the number casting down to the consumer level
* probably more that I cannot think of.

Reply:
Can you clarify how the following option would result in fewer types?

> define ESQLLiteral as union of types (based on values I'd say 2 subTypes should be enough)

I like the idea, but as far as I can tell that would actually result in three types: `ESQLNumberLiteral`, `ESQLStringLiteral` `ESQLLiteral` which seems to work against what you're asking here.

Reply:
Only the `ESQLLiteral` type will be extracted and the others will be filtered as tagged union.